x-pack/test/epm_api_integration/apis/template.ts

```diff
@@ -21,9 +21,11 @@ export default function({ getService }: FtrProviderContext) {
       name: templateName,
       body,
     });
+    // Checks if template loading worked as expected
```

Comment:
@skh @jfsiii Added a code comment in this PR after you asked about the different checks. Like this no need to go through CI again ;-)

Reply:
Thank you! My next question would be: Why does this test if template loading works as expected?

Let me explain why this test bothers me.

It is not an api integration test. The "normal" setup for a test in this subdirectory is:
* ES server runs, as specified in `config.ts`
* Kibana server runs, as specified in `config.ts`
* The test suite runs in a totally different context (and different node process) and has no access to Kibana internals. It just happens to also be written in JS/TS, and be in the same repo.
The test suite code sends requests to the API, just like any other client (like the Kibana UI) would do, and checks if what comes back meets the specification.

This test does not follow that pattern. It is more like a unit test in that it imports a function directly from our plugin code, runs it, and tests its output for validity. It is different from a unit test that lives in the main plugin directory because it uses the running ES server to check if the generated artefact (here the generated template) is valid.

This is a useful test, and I'm not against doing it this way, even if it completely breaks through the abstraction boundaries that are normally true for API integration tests (testing suite code clearly separated from tested code). I also don't know a better way to do it -- technically this belongs into a unit test because it tests just one function, but in unit tests there's no ES server running, so it couldn't be used for validity checks like it's done here. Instead we'd need to use mock objects, which would need some effort, and not necessarily give better results -- especially as our mock objects will not know when ES changes in a way that makes our artefacts invalid.

Ideally we would test this during real API integration tests, but as there is no API triggering this code path yet (it will be called from the data source API, eventually), we can't add those now.

So I'm overall fine with this method, but I'd like to see a comment with a clear explanation what is happening here, and why.

@jfsiii what do you think?

Reply:
Also, please document the assumption that ES will complain as soon as the template is invalid. Is this true for all variants of "invalid"? Could we construct a faulty template that ES will accept without an error?

Reply:
Agree that it is not the "usual" way to test such a thing in the context of KB but until we found a better way, lets keep it.

I'm happy to add more documentation parts here if needed, at the same time I would argue this is just the basic behaviour of the ES REST API documented here: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates.html Would be a link to this be enough? For the faulty template: It is easy to create an invalid template and get an error back. Creating a faulty template which is accepted will be harder, if it is possible, it would be a bug in ES. But this is not the purpose of this test. All it does is checking if ES accepts our template and we rely on ES that it works as expected.

Reply:
The comments I'd like to see are 

`// This is not an API integration test.
// This test is run here because we do need a running ES server instance to check our templates are valid, and we don't have that in unit tests.`

`// Assume templates are valid if ES accepts them. Anything else would be an ES bug`

or variants thereof.

Reply:
@skh Got it, thanks for providing an example. Will adjust it accordingly.

Reply:
> Got it, thanks for providing an example. Will adjust it accordingly.

Also, don't let these comments keep you from merging, we can add them later.